File: examples/simple-java/src/main/java/example/PersonMapping.java
package example;

import dev.akkinoc.spring.boot.orika.OrikaMapperFactoryConfigurer;
import ma.glasnost.orika.MapperFactory;
import org.springframework.stereotype.Component;

@Component
public class PersonMapping implements OrikaMapperFactoryConfigurer {

    @Override
    public void configure(MapperFactory orikaMapperFactory) {
        orikaMapperFactory.classMap(PersonSource.class, PersonDestination.class)
                .field("firstName", "givenName")
                .field("lastName", "sirName")
                .byDefault()
                .register();
    }

}

File: examples/simple-kotlin/src/main/kotlin/example/PersonMapping.kt
package example

import dev.akkinoc.spring.boot.orika.OrikaMapperFactoryConfigurer
import ma.glasnost.orika.MapperFactory
import org.springframework.stereotype.Component

@Component
class PersonMapping : OrikaMapperFactoryConfigurer {

    override fun configure(orikaMapperFactory: MapperFactory) {
        orikaMapperFactory.classMap(PersonSource::class.java, PersonDestination::class.java)
                .field("firstName", "givenName")
                .field("lastName", "sirName")
                .byDefault()
                .register()
    }

}
